src/libultra/io/aisetnextbuf.c

```diff
@@ -3,29 +3,29 @@
 //! Note that this is not the same as the original libultra
 //! osAiSetNextBuffer, see comments in the function
 
-s32 osAiSetNextBuffer(void* buf, u32 size) {
+s32 osAiSetNextBuffer(void* buf, size_t size) {
     static u8 D_80130500 = false;
-    u32 bufAdjusted = (u32)buf;
+    uintptr_t bufAdjusted = (uintptr_t)buf;
     s32 status;
 
     if (D_80130500) {
-        bufAdjusted = (u32)buf - 0x2000;
+        bufAdjusted = (uintptr_t)buf - 0x2000;
     }
-    if ((((u32)buf + size) & 0x1FFF) == 0) {
+    if ((((uintptr_t)buf + size) & 0x1FFF) == 0) {
         D_80130500 = true;
     } else {
         D_80130500 = false;
     }
 
     // Originally a call to __osAiDeviceBusy
-    status = HW_REG(AI_STATUS_REG, s32);
+    status = HW_REG(AI_STATUS_REG, uintptr_t);
```

Comment:
To pick one that leapt out at me, this is a set of flags, as we see in the next line, so should be `u32`. I suspect there are many more of these. You probably shouldn't be touching libultra until @Thar0 's finished his cleanup anyway.

Reply:
I realized this is resolved, but this is really just a bad usage of a fake macro. HW_REG is not a real libultra macro, and should be replaced with IO_WRITE here and elsewhere.